File: app/src/main/java/net/xcreen/restsms/AboutAppFragment.java
package net.xcreen.restsms;

import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;

import androidx.core.app.ActivityCompat;
import androidx.fragment.app.Fragment;

import android.provider.Telephony;
import android.telephony.TelephonyManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class AboutAppFragment extends Fragment {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_about_app, container, false);

        //Set Version-Name
        TextView versionResultTV = rootView.findViewById(R.id.about_app_version_result_textview);
        versionResultTV.setText(BuildConfig.VERSION_NAME);

        //Set Version
        TextView versionNameResultTV = rootView.findViewById(R.id.about_app_versionname_result_textview);
        versionNameResultTV.setText(String.valueOf(BuildConfig.VERSION_CODE));

        //Check if has SMS-Permission
        TextView smsPermissionResultTV = rootView.findViewById(R.id.about_app_smspermission_result_textview);
        try {
            if (ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.SEND_SMS) == PackageManager.PERMISSION_GRANTED) {
                smsPermissionResultTV.setText(R.string.yes);
            } else {
                smsPermissionResultTV.setText(R.string.no);
            }
        }
        catch (Exception ex){
            smsPermissionResultTV.setText(R.string.unknown);
            ex.printStackTrace();
        }

        //Check Sim-State
        int primarySim = TelephonyManager.SIM_STATE_ABSENT;
        int secondarySim = TelephonyManager.SIM_STATE_ABSENT;
        try {
            TelephonyManager telephonyManager = (TelephonyManager) getContext().getSystemService(Context.TELEPHONY_SERVICE);
            primarySim = telephonyManager.getSimState();
            if (Build.VERSION.SDK_INT >= 26) {
                primarySim = telephonyManager.getSimState(0);
                secondarySim = telephonyManager.getSimState(1);
            }
        }
        catch (Exception ex){
            ex.printStackTrace();
        }

        //Show Sim-State
        TextView simState1ResultTV = rootView.findViewById(R.id.about_app_sim1state_result_textview);
        int primarySimTextID = getTextIdBySimState(primarySim);
        if(primarySimTextID != 0) {
            simState1ResultTV.setText(primarySimTextID);
        }
        TextView simState2ResultTV = rootView.findViewById(R.id.about_app_sim2state_result_textview);
        int secondarySimTextID = getTextIdBySimState(secondarySim);
        if(secondarySimTextID != 0) {
            simState2ResultTV.setText(secondarySimTextID);
        }

        //Show Default-SMS-App
        String defaultSMSApp = Telephony.Sms.getDefaultSmsPackage(getContext());
        TextView defaultSMSAppResultTV = rootView.findViewById(R.id.about_app_defaultsmsapp_result_textview);
        if(defaultSMSApp != null){
            defaultSMSAppResultTV.setText(defaultSMSApp);
        }

        return rootView;
    }

    /**
     * Returns the String-ID for the Sim-Status
     * @param simState - SimState
     * @return id - Text-ID
     */
    private int getTextIdBySimState(int simState){
        switch (simState){
            case TelephonyManager.SIM_STATE_ABSENT:
                return R.string.error_sim_absent;
            case TelephonyManager.SIM_STATE_PIN_REQUIRED:
                return R.string.error_sim_pin;
            case TelephonyManager.SIM_STATE_PUK_REQUIRED:
                return R.string.error_sim_puk;
            case TelephonyManager.SIM_STATE_NETWORK_LOCKED:
                return R.string.error_sim_network_pin;
            case TelephonyManager.SIM_STATE_READY:
                return R.string.error_sim_ready;
            case TelephonyManager.SIM_STATE_NOT_READY:
                return R.string.error_sim_not_ready;
            case TelephonyManager.SIM_STATE_PERM_DISABLED:
                return R.string.error_sim_disabled;
            case TelephonyManager.SIM_STATE_CARD_IO_ERROR:
                return R.string.error_sim_card_error;
            case TelephonyManager.SIM_STATE_CARD_RESTRICTED:
                return R.string.error_sim_restricted;
            default:
                return 0;
        }
    }
}

File: app/src/main/java/net/xcreen/restsms/fragments/AboutAppFragment.kt
package net.xcreen.restsms.fragments

import android.Manifest
import android.content.Context
import android.content.pm.PackageManager
import android.graphics.PorterDuff
import android.os.Build
import android.os.Bundle
import android.provider.Telephony
import android.telephony.TelephonyManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.app.ActivityCompat
import androidx.fragment.app.Fragment
import net.xcreen.restsms.BuildConfig
import net.xcreen.restsms.R

class AboutAppFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_about_app, container, false)
        //Set Version-Name
        val versionResultTV = rootView.findViewById<TextView>(R.id.about_app_version_result_textview)
        versionResultTV.text = BuildConfig.VERSION_NAME
        //Set Version
        val versionNameResultTV = rootView.findViewById<TextView>(R.id.about_app_versionname_result_textview)
        versionNameResultTV.text = BuildConfig.VERSION_CODE.toString()
        //Check if has SMS-Permission
        val smsPermissionResultImageView = rootView.findViewById<ImageView>(R.id.about_app_smspermission_result_iv)
        try {
            smsPermissionResultImageView.setColorFilter(requireContext().getColor(R.color.colorError), PorterDuff.Mode.SRC_ATOP)
            if (ActivityCompat.checkSelfPermission(requireContext(), Manifest.permission.SEND_SMS) == PackageManager.PERMISSION_GRANTED) {
                smsPermissionResultImageView.setImageDrawable(requireContext().getDrawable(R.drawable.check_yes))
                smsPermissionResultImageView.setColorFilter(requireContext().getColor(R.color.colorPrimary), PorterDuff.Mode.SRC_ATOP)
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        //Check Sim-State
        var primarySim = TelephonyManager.SIM_STATE_ABSENT
        var secondarySim = TelephonyManager.SIM_STATE_ABSENT
        try {
            val telephonyManager = requireContext().getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager
            primarySim = telephonyManager.simState
            if (Build.VERSION.SDK_INT >= 26) {
                primarySim = telephonyManager.getSimState(0)
                secondarySim = telephonyManager.getSimState(1)
            }
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
        //Show Sim-State
        val simState1ResultTV = rootView.findViewById<TextView>(R.id.about_app_sim1state_result_textview)
        val primarySimTextID = getTextIdBySimState(primarySim)
        if (primarySimTextID != 0) {
            simState1ResultTV.setText(primarySimTextID)
        }
        val simState2ResultTV = rootView.findViewById<TextView>(R.id.about_app_sim2state_result_textview)
        val secondarySimTextID = getTextIdBySimState(secondarySim)
        if (secondarySimTextID != 0) {
            simState2ResultTV.setText(secondarySimTextID)
        }
        //Show Default-SMS-App
        val defaultSMSApp = Telephony.Sms.getDefaultSmsPackage(context)
        val defaultSMSAppResultTV = rootView.findViewById<TextView>(R.id.about_app_defaultsmsapp_result_textview)
        if (defaultSMSApp != null) {
            defaultSMSAppResultTV.text = defaultSMSApp
        }
        return rootView
    }

    /**
     * Returns the String-ID for the Sim-Status
     * @param simState - SimState
     * @return id - Text-ID
     */
    private fun getTextIdBySimState(simState: Int): Int {
        return when (simState) {
            TelephonyManager.SIM_STATE_ABSENT -> R.string.error_sim_absent
            TelephonyManager.SIM_STATE_PIN_REQUIRED -> R.string.error_sim_pin
            TelephonyManager.SIM_STATE_PUK_REQUIRED -> R.string.error_sim_puk
            TelephonyManager.SIM_STATE_NETWORK_LOCKED -> R.string.error_sim_network_pin
            TelephonyManager.SIM_STATE_READY -> R.string.error_sim_ready
            TelephonyManager.SIM_STATE_NOT_READY -> R.string.error_sim_not_ready
            TelephonyManager.SIM_STATE_PERM_DISABLED -> R.string.error_sim_disabled
            TelephonyManager.SIM_STATE_CARD_IO_ERROR -> R.string.error_sim_card_error
            TelephonyManager.SIM_STATE_CARD_RESTRICTED -> R.string.error_sim_restricted
            else -> 0
        }
    }
}